File: CS/CachingService/CachingService.asmx.cs
using System;
using DevExpress.Xpo.DB;
using System.Web.Services;
using System.ComponentModel;
using DevExpress.Xpo.DB.Helpers;

namespace DXSample.Service {
    [WebService(Namespace = "http://devexpress.example/")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    [ToolboxItem(false)]
    public class CachingService : WebService {
        [WebMethod]
        public DataCacheModificationResult ModifyData(DataCacheCookie cookie,
            ModificationStatement[] dmlStatements) {
            return ((DataCacheRoot)Application["DataStore"]).ModifyData(cookie, dmlStatements);
        }

        [WebMethod]
        public DataCacheResult NotifyDirtyTables(DataCacheCookie cookie,
            params string[] dirtyTablesNames) {
            return ((DataCacheRoot)Application["DataStore"]).NotifyDirtyTables(cookie, dirtyTablesNames);
        }

        [WebMethod]
        public DataCacheResult ProcessCookie(DataCacheCookie cookie) {
            return ((DataCacheRoot)Application["DataStore"]).ProcessCookie(cookie);
        }

        [WebMethod]
        public DataCacheSelectDataResult SelectData(DataCacheCookie cookie, SelectStatement[] selects) {
            return ((DataCacheRoot)Application["DataStore"]).SelectData(cookie, selects);
        }
    }
}

File: VB/CachingService/CachingService.asmx.vb
Imports DevExpress.Xpo.DB
Imports System.Web.Services
Imports System.ComponentModel
Imports DevExpress.Xpo.DB.Helpers

Namespace DXSample.Service

    <WebService([Namespace]:="http://devexpress.example/")>
    <WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)>
    <ToolboxItem(False)>
    Public Class CachingService
        Inherits WebService

        <WebMethod>
        Public Function ModifyData(ByVal cookie As DataCacheCookie, ByVal dmlStatements As ModificationStatement()) As DataCacheModificationResult
            Return CType(Application("DataStore"), DataCacheRoot).ModifyData(cookie, dmlStatements)
        End Function

        <WebMethod>
        Public Function NotifyDirtyTables(ByVal cookie As DataCacheCookie, ParamArray dirtyTablesNames As String()) As DataCacheResult
            Return CType(Application("DataStore"), DataCacheRoot).NotifyDirtyTables(cookie, dirtyTablesNames)
        End Function

        <WebMethod>
        Public Function ProcessCookie(ByVal cookie As DataCacheCookie) As DataCacheResult
            Return CType(Application("DataStore"), DataCacheRoot).ProcessCookie(cookie)
        End Function

        <WebMethod>
        Public Function SelectData(ByVal cookie As DataCacheCookie, ByVal selects As SelectStatement()) As DataCacheSelectDataResult
            Return CType(Application("DataStore"), DataCacheRoot).SelectData(cookie, selects)
        End Function
    End Class
End Namespace
